Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -59,19 +59,28 @@ def edit
           "#{path} doesn't exist on disk."
         end
 
-        message = if args.cask?
-          <<~EOS
-            #{not_exist_message}
-            Run #{Formatter.identifier("brew create --cask --set-name #{path.basename(".rb")} $URL")} \
-            to create a new cask!
-          EOS
+        suggestion_message = if args.cask?
+          if !CoreCaskTap.instance.installed? && Homebrew::API::Cask.all_casks.key?(path.basename(".rb").to_s)
+            command = "brew tap --force #{CoreCaskTap.instance.name}"
+            "Run #{Formatter.identifier(command)} to tap #{CoreCaskTap.instance.name}!"
+          else
+            "Run #{Formatter.identifier("brew create --cask --set-name #{path.basename(".rb")} $URL")} " \
+              "to create a new cask!"
+          end
         else
-          <<~EOS
-            #{not_exist_message}
-            Run #{Formatter.identifier("brew create --set-name #{path.basename} $URL")} \
-            to create a new formula!
-          EOS
+          if !CoreTap.instance.installed? && Homebrew::API::Formula.all_formulae.key?(path.basename.to_s)
```

Comment:
```suggestion
          if !CoreTap.instance.installed? && Homebrew::API::Formula.all_formulae.key?(path.basename(".rb").to_s)
```

Reply:
For some reason, `path` is printing a full path for casks but just a name for formulae: https://github.com/Homebrew/brew/pull/13120

Reply:
It has to do with this code block in `NamedArgs#to_paths`.

https://github.com/Homebrew/brew/blob/62c2c335e56f29e034cb26348c32bfcf9b43b80c/Library/Homebrew/cli/named_args.rb#L243-L254

Basically, we don't check if a path exists if either `--formula` or `--cask` is specified. Otherwise, we check both paths and then just return the string as a path if neither exists.

Reply:
For example, with the `--formula` flag.

```sh
~ $ brew edit --formula xpsasth
...
Error: Invalid usage: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xpsasth.rb doesn't exist on disk.
Run brew create --set-name xpsasth.rb $URL to create a new formula!
```